### ISSUE TITLE
hiptest-publisher install on OSX

### ISSUE THREAD

**Issue by apowers313** [hiptest/hiptest-publisher#22]
I'm documenting this here for those that might suffer from the same issues I did installing hiptest-publisher on OSX.

<hr>
tl;dr: run these commands:

> export NOKOGIRI_USE_SYSTEM_LIBRARIES=1
> export ARCHFLAGS="-arch x86_64"
> gem install hiptest-publisher --user-install -n~/bin

<hr>

Problem 1 - libxml2 is missing.

The symptom looks something like this:

> checking for xmlParseDoc() in -llibxml2... no
> libxml2 is missing.  Please locate mkmf.log to investigate how it is failing.
> *** extconf.rb failed ***
> [ garbage scrolls on and on ]

Note that for me it wasn't actually missing, it was failing to link because it was trying to link against the 'i386' version instead of the 'x86_64' version. Setting the environment variable `ARCHFLAGS="-arch x86_64"` fixes this problem.

<hr>

Problem 2: still failing

At this point the build is still failing with `libxml2 missing`, but looking through the logs it has different compile problems. Unfortunately I don't have the logs, but the fix was `NOKOGIRI_USE_SYSTEM_LIBRARIES=1`

<hr>

Problem 3: Permission denied

Now it's trying to install a binary to a bad directory:

> Building native extensions.  This could take a while...
> ERROR:  While executing gem ... (Errno::EPERM)
> Operation not permitted - /usr/bin/nokogiri

The fix for this is to install the gem locally instead of globally using the command `gem install hiptest-publisher --user-install -n~/bin`

**Comment by vincent-psarga** [comment]
Thanks for the feedback :)
Just a quick question, do you use RVM or the Ruby installed with OSX ? I never had problem installing hiptest-publisher on OSX as I'm using RVM.
If you don't mind, I'll include your description of the issues in the next release documentation :)

**Comment by apowers313** [comment]
I think just the vanilla OSX Ruby (unless MacPorts installed RVM while I wasn't looking).

Happy to have you include my comments.

**Comment by vincent-psarga** [comment]
The documentation has been added in the last release.
Thanks again for your contribution :)
